src/app/auth.guard.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router, CanActivate } from '@angular/router';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { AuthorizationService } from './features/login/services/authorization.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+
+  constructor(
+    private authorizationService: AuthorizationService,
+    private router: Router) { }
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.authorizationService.isAuthenticated()
+      .pipe(
+        map(isAuthenticated => {
+          if (isAuthenticated) {
+            return true;
+          } else {
+            this.router.navigate(['/login']);
```

Comment:
in this case you should return `UrlTree`

Reply:
you can make smth like this:
 ```
isAuthenticated: addType => !!isAuthenticated || this.router.createUrlTree([/login])
```

Reply:
Minor: pls create enum with all routes, it's make easer feature cahnges

Reply:
1. Changed to return UlrTree
2. created enums with routes